nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/CountTextTest.java

```diff
@@ -0,0 +1,316 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.util.MockComponentLog;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.Security;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+
+public class CountTextTest {
+    private static final String TLC = "text.line.count";
+    private static final String TLNEC = "text.line.nonempty.count";
+    private static final String TWC = "text.word.count";
+    private static final String TCC = "text.character.count";
+
+    @BeforeAll
+    static void setUpOnce() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    @Test
+    void testShouldCountAllMetrics() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(CountText.class);
+
+        runner.setProperty(CountText.TEXT_LINE_COUNT_PD, "true");
+        runner.setProperty(CountText.TEXT_LINE_NONEMPTY_COUNT_PD, "true");
+        runner.setProperty(CountText.TEXT_WORD_COUNT_PD, "true");
+        runner.setProperty(CountText.TEXT_CHARACTER_COUNT_PD, "true");
+
+        final Path inputPath = Paths.get("src/test/resources/TestCountText/jabberwocky.txt");
+
+        final Map<String, String> expectedValues = new HashMap<>();
+        expectedValues.put(TLC, "34");
+        expectedValues.put(TLNEC, "28");
+        expectedValues.put(TWC, "166");
+        expectedValues.put(TCC, "900");
+
+        runner.enqueue(Files.readAllBytes(inputPath));
+
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(CountText.REL_SUCCESS, 1);
+        FlowFile flowFile = runner.getFlowFilesForRelationship(CountText.REL_SUCCESS).get(0);
+        for (final Map.Entry<String, String> entry: expectedValues.entrySet()) {
+            final String attribute = entry.getKey();
+            final String expectedValue = entry.getValue();
+
+            assertEquals(expectedValue, flowFile.getAttribute(attribute));
+        }
```

Comment:
This approach should be replaced with `MockFlowFile.assertAttributeEquals()`, which would also remove the need for the `Map` of expected values.

Reply:
Making the changes